### PR TITLE
Class dependencies

### PR DESCRIPTION
**Pull request by delijati**
Define dependencies class wide :) Sorry for the formatting changes.

Example:

```
@controller.resource()
class User:
    dependencies = [Depends(user_exists)]

    @controller.route.get("/users/{user_id}", dependencies=[Depends(user_is_id)])
    def read_users(self, user_id: int):
        return {"user_id": user_id}
```

### CODE REVIEW

**Comment by KiraPC** [comment]
@delijati 

Good feature, but, can you add some documentation that explain it? Maybe in the readme?

I prefer single quote (') instead of double (")

Could you use them?

**Comment by delijati** [comment]
> @delijati
> 
> Good feature, but, can you add some documentation that explain it? Maybe in the readme?

Will do.

> I prefer single quote (') instead of double (")
> Could you use them?

I would rather not ecpecially not for doc strings -> https://www.python.org/dev/peps/pep-0008/#id25 And almost every tool uses double qoutes. I'm just a lazy bastard and use automation for formatting ;) See https://github.com/KiraPC/fastapi-router-controller/issues/17

**Comment by KiraPC** [comment]
@delijati 

Ok, let's use double.

I started with single because I came from JS :)

Anyway, I will think for automation formatting.

File: fastapi_router_controller/lib/controller.py [hunk 1]
```diff
@@ -88,47 +92,56 @@ def wrapper(cls):
...
+
         for route in router.routes:
             # get the signature of the endpoint function
             signature = inspect.signature(route.endpoint)
             # get the parameters of the endpoint function
             signature_parameters = list(signature.parameters.values())
+            # add class dependencies
+            if dependencies:
+                for depends in dependencies[::-1]:
+                    route.dependencies.insert(0, depends)
```
Comment (KiraPC) [inline fastapi_router_controller/lib/controller.py:115]: @delijati Does route.dependencies always exists?

Comment (KiraPC) [reply fastapi_router_controller/lib/controller.py:115]: @delijati I will move it at line 106, just to keep it separated from rewrite sign

**Review by KiraPC — CHANGES_REQUESTED** [review]
@delijati Please add docs, also in class doc string, before merging it

File: fastapi_router_controller/lib/controller.py (hunk 1 above)
Comment (delijati) [reply fastapi_router_controller/lib/controller.py:115]: yes https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L342 and sure

**Comment by delijati** [comment]
Added some docs. I made the example on the frontpage copy pastable ;)

**Review by KiraPC — APPROVED** [review]

**Pull request merged**